deployment/common/changeset/test_helpers.go

```diff
@@ -0,0 +1,87 @@
+package changeset
+
+import (
+	"fmt"
+	"testing"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
+	jobv1 "github.com/smartcontractkit/chainlink-protos/job-distributor/v1/job"
+	"github.com/smartcontractkit/chainlink-testing-framework/lib/utils/testcontext"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+type ChangesetApplication struct {
+	Changeset deployment.ChangeSet[any]
+	Config    any
+}
+
+func WrapChangeSet[C any](fn deployment.ChangeSet[C]) func(e deployment.Environment, config any) (deployment.ChangesetOutput, error) {
+	return func(e deployment.Environment, config any) (deployment.ChangesetOutput, error) {
+		var zeroC C
+		if config != nil {
+			c, ok := config.(C)
+			if !ok {
+				return deployment.ChangesetOutput{}, fmt.Errorf("invalid config type, expected %T", c)
+			}
+			return fn(e, config.(C))
+		}
+
+		return fn(e, zeroC)
+	}
+}
+
+// ApplyChangesets applies the changeset applications to the environment and returns the updated environment.
+func ApplyChangesets(t *testing.T, e deployment.Environment, timelocksPerChain map[uint64]*gethwrappers.RBACTimelock, changesetApplications []ChangesetApplication) (deployment.Environment, error) {
+	currentEnv, err := e.Copy()
+	if err != nil {
+		return e, fmt.Errorf("failed to copy environment: %w", err)
+	}
+	for i, csa := range changesetApplications {
+		out, err := csa.Changeset(currentEnv, csa.Config)
+		if err != nil {
+			return e, fmt.Errorf("failed to apply changeset at index %d: %w", i, err)
+		}
+		if out.AddressBook != nil {
+			err := currentEnv.ExistingAddresses.Merge(out.AddressBook)
```

Comment:
we're planning on making ExistingAddresses read-only as per CCIP-4247. So I'd invert this:
newAddresses = out.AddressBook.Merge(currentEnv.ExistingAddresses)

Reply:
There is an issue with this. The consecutive changesets will load the state and try to look for dependent contracts. that will fail if we do not update the existing addresses.

Reply:
Discussed offline and addressed